lib/util.js

```diff
@@ -304,7 +304,11 @@ function formatValue(ctx, value, recurseTimes) {
       return ctx.stylize(RegExp.prototype.toString.call(value), 'regexp');
     }
     if (isDate(value)) {
-      return ctx.stylize(Date.prototype.toISOString.call(value), 'date');
+      if (value.toString() === 'Invalid Date') {
```

Comment:
@bnoordhuis well, it means keeping _an_ extra date object once. It can be kept in a utility and used everywhere invalid dates are needed :P 

I'm fine with `Number.isNaN(value.getTime(), NaN)` although it's not obvious to readers who are not familiar with this obscure detail.